core/service/manager.go

```diff
@@ -46,10 +46,11 @@ type Service interface {
 	Start(providerID identity.Identity) (dto_discovery.ServiceProposal, session.ConfigProvider, error)
 	Wait() error
 	Stop() error
+	GetType() string
```

Comment:
This looks artificial. Maybe its better to get type from serviceProposal returned by manager.service.Start(providerID) if needed? We should know what we are running.

Reply:
The way I see it, we'll still need the manager(maybe the naming needs a change, but naming is hard). I like that it's responsible for 
1) Registering/pinging the proposal
1) Starting the service it's given
1) Starting to listen for the relevant dialogs

We'll need these for every service, so in my head the component makes perfect sense.

Reply:
In that case yes - better naming is needed. I didn't expect to see multiple managers :)